main.js

```diff
@@ -109,6 +96,28 @@ function displayLocalStorageCards() {
     for (var i = 0; i < fromStorage.length; i++) {
       createIdea(fromStorage[i].id, fromStorage[i].title, fromStorage[i].body, fromStorage[i].starred)
     }
+    setFavoriteStyle(window);
+  }
+}
+
+function checkCardID(event) {
+  var cardId = JSON.parse(event.path[2].dataset.id);
+  var card = null;
+  for (var i = 0; i < ideaArray.length; i++) {
+    if (cardId === ideaArray[i].id) {
+      card = ideaArray[i];
+    }
+  }
+  return card;
+}
+
+function setFavoriteStyle(window) {
```

Comment:
This is a pretty creative solution!

I think there might be something easier though. Is there a way that you could apply the `.active-favorite-btn` class in your `showIdea` function? That way, the favorite would apply when you render each card to the DOM and there wouldn't be any need for this function. 

Just some food for thought 🙂

Additionally, do you need to pass `window` to this function? Where is window defined in your application?

Reply:
I see what you mean about the window function. I'll remove that and test it. 

I get what you mean about leveraging the showIdea function for the favoriting, but I don't know if I could figure it out in this deadline.

Reply:
I totally get that! Feel free to ping me on Slack after your deadline and I'll walk you through it 🙂